src/theory/strings/word.h

```diff
@@ -0,0 +1,119 @@
+/*********************                                                        */
+/*! \file word.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Utility functions for words.
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__STRINGS__WORD_H
+#define CVC4__THEORY__STRINGS__WORD_H
+
+#include <vector>
+
+#include "expr/node.h"
+
+namespace CVC4 {
+namespace theory {
+namespace strings {
+namespace word {
```

Comment:
Why introduce a separate namespace?

Reply:
Same reason as we have `util::`.  These functions would otherwise pollute the strings namespace significantly.  Also, I think it's nice to have this so that there is a consistent naming scheme when calling them.

Reply:
An alternative would be to make them static members of a class?

Reply:
I think that I'd prefer that, yes. I like to keep the namespaces and directories in sync.